File: UVA10038.py
while True:
    try:
        list1 = input().split(" ")
        list1 = [int(i) for i in list1]
        total = list1[0]
        list1 = list1[1:]
        dict1 = {}
        flag = True
        for i in range(total - 1):
            num = abs(list1[i] - list1[i + 1])
            if(dict1.get(num,0) != 0 or num >= total or num == 0):
                print("Not jolly")
                flag = False
                break
            else:
                dict1[num] = 1
        if flag:
            print("Jolly")
    except EOFError:
        break

File: UVA10038.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int total,num,num1,diff;
    bool flag = true;
    int times = 0;
    int array[3000];
    while(cin >> total)
    {
        flag = true;
        num = 0;
        for(int i = 0;i < total;i++)
        {
            cin >> array[i];
        }
        for(int i = 0; i < total - 1;i++)
        {
            num += abs(array[i] - array[i + 1]);
        }
        if(num == (total * (total - 1)) / 2)
        {
            cout << "Jolly" << endl;
        }
        else
        {
            cout << "Not jolly" << endl;
        }
    }
    return 0;
}